### ISSUE TITLE
"Other root directories have problems"

### ISSUE THREAD

**Issue by johnslemmer** [groboclown/p4ic4idea#164]
On the server connection page I'm getting:

> Other root directories have problems

What does this error even mean?

My best guess is that this has something to do with how my company uses Perforce Streams.

My company uses streams to handle our projects (and I'm sure we are using them in a weird way too potentially).  Long story short my perforce client looks something like this.

```
Stream:	//streams/a_single_stream

View:
	//streams/a_single_stream/... //my-perforce-client/...

```
However, the way we have setup our streams is that only ever one directory is inside a view.  The actual project directory.

```
$ cd my/perforce/client/root/dir/
$ ls
the-android-project/
```

Now when I want to open Android Studio I want to open the project at `my/perforce/client/root/dir/the-android-project`.  However, I believe this IDEA plugin may be expecting Android Studio to be opened at `my/perforce/client/root/dir/`.  Is this at all true?  Or is there something else entirely messed up?

macOS Sierra
Android Studio 2.3.3
p4ic4idea 0.9.5

**Comment by groboclown** [comment]
I've used the plugin with the same kind of stream setup that you're using, so that should be fine.

The message `Other root directories have problems` means that the server connection tab is showing you the output for one directory, but there's another directory in the drop-down that has connection problems.

This "multiple root directory" situation can happen when you setup your IDE to have multiple VCS roots, or if you have multiple p4 config files that have different server connection information.

Another potential source is a bug in the plugin ;) mis-identifying the root directories.

Please inspect the other root directories to see if this list makes sense.  If something looks unexpected, please let me know.

**Comment by johnslemmer** [comment]
Thanks for the reply @groboclown.

I actually was able to get it to work.

`.idea/vcs.xml` was checked into perforce with this:
```
  <component name="VcsDirectoryMappings">
    <mapping directory="$PROJECT_DIR$/.." vcs="p4ic" />
  </component>
```

Undoubtedly this was meant to point at the perforce client root (because of how we use streams and such).

However, when I left the configuration as above and go to the _Server Connection_ tab then the plugin only allows me to choose the Android project root from the dropdown as the _Root dir_.  Which then leads to the error described in my initial post: _Other root directories have problems._ . I'm assuming this happens because there is a mismatch between the VCS directory mapping and the root dir that the p4ic plugin is configured with (is that correct?).
 
Now, I have changed the VCS directory mappings to be (which sets up the Android project directory with VCS):
```
  <component name="VcsDirectoryMappings">
    <mapping directory="" vcs="p4ic" />
  </component>
```

However, my questions now is, will submitting this change affect my teammates at all?  Also, why did the previous configuration work for them (they are using Linux BTW)?  Also, is there a problem with the problem where it didn't allow me to choose the perforce client root when it was configured as a VCS directory mapping?

Anyway, I got it to work.  Hopefully somebody else finds this conversation useful.
